src/audio.c

```diff
@@ -643,9 +668,22 @@ static void check_telev(struct audio *a, struct autx *tx)
  */
 static void auplay_write_handler(void *sampv, size_t sampc, void *arg)
 {
-	struct aurx *rx = arg;
+	int err = 0;
+	struct audio *a = arg;
+	struct aurx *rx = &a->rx;
 	size_t num_bytes = sampc * aufmt_sample_size(rx->play_fmt);
 
+	err = stream_decode(a->strm);
+
+	while (err == EAGAIN ||
+			(!err && aubuf_cur_size(rx->aubuf) < num_bytes)) {
+
+		if (err == EAGAIN)
+			++rx->again;
+
+		err = stream_decode(a->strm);
+	}
+
```

Comment:
we should not do audio decode in the audio play handler.

this handler may be called from audio DSP realtime interrupt-handler.

I suggest instead to do decoding in:

1. a new thread
2. a `struct tmr` polling loop

Reply:
Ok, I also agree and will add a commit that does one of this suggestions.

Reply:
I think both with #ifdef HAVE_PTHREAD, right?

Found that the union around the "pthread_t tid" is obsolete because the AUDIO_MODE_TMR was removed. But I could use the original union approach to select tmr vs thread. Ok?